ash/src/extensions/khr/swapchain.rs

```diff
@@ -3,17 +3,25 @@ use super::DeviceGroup;
 use crate::prelude::*;
 use crate::vk;
 use crate::RawPtr;
-use crate::{Device, Instance};
+use crate::{Device, Entry, Instance};
 use std::ffi::CStr;
 use std::mem;
 
+/// <https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VK_KHR_swapchain.html>
 #[derive(Clone)]
 pub struct Swapchain {
     handle: vk::Device,
     fp: vk::KhrSwapchainFn,
 }
 
 impl Swapchain {
+    /// [`Instance`] functions cannot be loaded from a [`Device`] and will always panic when called:
+    /// - [`Self::get_physical_device_present_rectangles()`]
+    ///
+    /// Load this struct using an [`Instance`] instead via [`Self::new_from_instance()`], or
+    /// suppress the deprecation warning if the above [`Instance`] function is not called.
+    /// This will be solved in the next breaking `ash` release.
+    #[deprecated = "`Instance` functions loaded via this function will always panic. Load via `new_from_instance()` instead."]
```

Comment:
I'm afraid that this deprecation warning comes over too aggressively on downstream consumers when we release `0.37.4`, pretty much everyone using Vulkan will be using a `swapchain` and have their project fail linting.

On the other hand I really want to make users aware of this deficiency, though they'll likely run into the panic anyway whenever testing the codepath of the instance function (if anyone calls into it at all - we received _zero_ reports about these) and hopefully find this elaborate explanation to understand the issue (or we can guide them in case an issue gets created). It'll anyway be solved properly in `0.38`, which I hope to release "soon™" after processing all the known breaking changes and actually finalizing what the proper solution (#734) is going to look like!

Reply:
If this is, as the diff suggests, only an issue for `get_physical_device_present_rectangles`, maybe it's not that important to call out? That function seems *very* niche.

Reply:
`get_device_group_surface_present_modes2` seems very niche as well, so I'll drop `#[deprecated]` on all these.